rundeckapp/grails-app/services/rundeck/services/JobSchedulerService.groovy

```diff
@@ -137,4 +214,80 @@ class QuartzJobScheduleManager implements JobScheduleManager {
         false
     }
 
+    /**
+     * Removes PENDING triggers if their job detail indicates they are over 2 minutes old.
+     * This allows for some unexpected delays in committing due to lock timeouts and etc.
+     */
+    private void cleanupTriggers() {
+        GroupMatcher<TriggerKey> matcher = GroupMatcher.groupEquals(TRIGGER_GROUP_PENDING)
+        quartzScheduler.getTriggerKeys(matcher).each { triggerKey ->
+            if (triggerKey.group == TRIGGER_GROUP_PENDING) {
+                Trigger trigger = quartzScheduler.getTrigger(triggerKey)
+                Map data = trigger.jobDataMap
+                Instant created = data.get('meta.created') as Instant
+
+                if (created.isBefore(created.minus(2, ChronoUnit.MINUTES)))
+                    quartzScheduler.unscheduleJob(triggerKey)
+            }
+        }
+    }
+
+    /**
+     * If an execution has been re-assigned to the local node we reschedule the PENDING job
+     */
+    @Subscriber
+    void afterUpdate(PostUpdateEvent event) {
+        if(!(event.entityObject instanceof Execution))
+            return
+
+        Execution execution = event.entityObject as Execution
+
+        if (isExecutionReassignedToLocal(execution)) {
```

Comment:
should this also check `isExecutionTypeUser` and `isExecutionStatusScheduled` just like the afterInsert method does?

Reply:
@ProTip @sjrd218